### PR TITLE
Bump plasma-contracts SHA to 1c7cd867

### LINKED ISSUE #1138: Bump plasma-contracts SHA to Nov 12
Target SHA: https://github.com/omisego/plasma-contracts/commit/1c7cd867b1353ae72c825a6d15b2d382689f30ad

A bit of work to change the deployment interface as @boolafish pointed out:

- https://github.com/omisego/plasma-contracts/commit/db824055bb3037d2fc1034b8ec751494cbe1dd4b
- https://github.com/omisego/plasma-contracts/commit/43968f6fdab815fb5b46cfcf2d9803d3a26ea1e9

Build errors: https://circleci.com/workflow-run/856730f1-ada9-4c4c-9017-53a00806b033

TODOs:
- [x] Bump https://github.com/omisego-images/docker-elixir-omg/blob/master/tester/Dockerfile.plasma_deployer#L15
- [ ] Bump `elixir-omg` to new SHA
- [ ] Fix `elixir-omg` contracts deployment errors during tests (align with the 2 plasma-contracts commits above)
- [ ] Let devops know when `elixir-omg` PR with new contracts SHA is merged.

### PR DESCRIPTION
**Pull request by unnawut**
Closes #1138

## Overview

Upgrade plasma-contracts to SHA `1c7cd867` and related changes to align with the updated contract interfaces.

## Changes

- Upgrade plasma-contracts to `1c7cd867b1353ae72c825a6d15b2d382689f30ad`
- Add `OMG.Eth.Encoding.encode_constructor_params/1`
- Add `OMG.Eth.Encoding.ContractConstructor.extract_params/1`

Align with latest plasma-contracts' interfaces
- Start standard exits uses `uint256` for `utxo_pos`
- `EthDepositVerifier` deployment requires `transaction_type` and `output_type`
- `Erc20DepositVerifier` deployment requires `transaction_type` and `output_type`
- `EthVault` deployment requires `safe_gas_stipend`
- `Erc20Vault` deployment requires `safe_gas_stipend`
- `PaymentExitGame` deployment requires `safe_gas_stipend`

## Testing

- `mix test` should pass
- `mix test --only common` should pass
- `mix test --integration` should pass
- `docker-compose up` should be successful

### CODE REVIEW

File: apps/omg_eth/lib/omg_eth/encoding.ex [hunk 1]
```diff
@@ -50,4 +50,39 @@ defmodule OMG.Eth.Encoding do
...
+    |> ABI.TypeEncoder.encode_raw(types)
+    # NOTE: we're not using `to_hex` because the `0x` will be appended to the bytecode already
+    |> Base.encode16(case: :lower)
+  end
+
+  @doc """
+  Reduces a list of smart contract constructor parameters into a tuple of types list
+  and values list, so that it can be passed into `ABI.TypeEncoder.encode_raw/1`.
+  """
+  @spec reduce_constructor_params(list()) :: {list(), list()}
```
Comment (pdobacz) [inline apps/omg_eth/lib/omg_eth/encoding.ex]: maybe make this private and test via `encode...`. Looks like a worker function, it's only ever used in `encode` and tests

Comment (unnawut) [reply apps/omg_eth/lib/omg_eth/encoding.ex]: I originally made it defp but it would make the tests less comprehensible so I don't know... :thinking:

Comment (unnawut) [reply apps/omg_eth/lib/omg_eth/encoding.ex]: But actually.. I can decode then asserts. I'll update with your suggestions.

Comment (InoMurko) [reply apps/omg_eth/lib/omg_eth/encoding.ex]: If you want to test it separately move it to a separate module 😉

Comment (unnawut) [reply apps/omg_eth/lib/omg_eth/encoding.ex]: @InoMurko Not sure if I get what you meant. Like this? https://github.com/omisego/elixir-omg/pull/1139/commits/29f6ffb2d45225a7e6a91ca68ad139c219bd171c

Comment (InoMurko) [reply apps/omg_eth/lib/omg_eth/encoding.ex]: bingo :)

**Comment by unnawut** [comment]
All green. Will merge on Monday morning BKK time so no breakage over the weekend.

**Pull request merged**